vxquery-core/src/main/javacc/xquery-grammar.jj

```diff
@@ -1945,10 +1951,85 @@ ASTNode Constructor()  :
     (
         result = DirectConstructor()
         | result = ComputedConstructor()
+        | result = JsonConstructor()
     ) {
         return result;
     }
 }
+ASTNode JsonConstructor() :
+{
+    ASTNode result;
+}
+{
+	(result = ObjectConstructor() | result = ArrayConstructor()){
+		return result;
+	}
+}
+
+ASTNode ObjectConstructor() :
+{
+    List<ASTNode> content=new ArrayList<ASTNode>();
+    Token t;
+    ASTNode pc;
+}
+{
+    t=<LbraceExprEnclosure>
+    (pc=PairConstructor() {content.add(pc);} ( "," pc=PairConstructor() {content.add(pc);})* )
+    <Rbrace>
+    {
+        ObjectConstructor obj=new ObjectConstructor(createSourceLocation(t));
+        obj.setContent(content);
+        return obj;
+    }
+
+}
+
+ASTNode PairConstructor() :
+{
+    String key;
+    ASTNode value;
+    Token t;
+}
+{
+    t= <StringLiteral> {key=t.image;}
+    (":" | ":?")
+    (value=Literal() | value=ArrayConstructor()){
```

Comment:
Instead of supporting Literal() and ArrayConstructor() here, we should just support ExprSingle() [1] to support all expressions.

[1] http://jsoniq.org/docs/JSONiqExtensionToXQuery/html-single/index.html#section-grammar-summary